pkg/patterns/declarative/reconciler.go

```diff
@@ -255,6 +231,21 @@ func (r *Reconciler) applyOptions(opts ...reconcilerOption) {
 	r.options = params
 }
 
+// Validate compatibility of selected options
+func (r *Reconciler) validateOptions() error {
+	var errs []string
+
+	if r.options.prune && r.options.labelMaker == nil {
+		errs = append(errs, "WithApplyPrune must be used with the WithLabels option")
```

Comment:
I had an earlier comment that we should error / warn if prune is used without labels ... but you covered it :thumbsup: